### PR TITLE
CLDR-14804 update Locales.txt based on research

### PR DESCRIPTION
**Pull request by srl295**
- remove * from all but active TCs
- remove lines that were unused
- ADD lines for locales that had votes and/or vetters

CLDR-14804

- [ ] This PR completes the ticket.

Data was generated using #1293

### CODE REVIEW

**Comment by btangmu** [comment]
"Caused by: java.lang.IllegalArgumentException: Cannot have default content locale in Locales.txt: google ; nb ; moderate ;"

File: tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/Locales.txt [hunk 1]
```diff
@@ -2,388 +2,282 @@
...
-IBM	;	zh_Hant	;	modern
-IBM	;	ja	;	modern
-IBM	;	ko	;	modern
-IBM	;	fr	;	modern
-IBM	;	de	;	modern
-IBM	;	it	;	modern
-IBM	;	es	;	modern
-IBM	;	pt	;	modern
-# IBM Group 2
+IBM	;	am	;	moderate
```
Comment (macchiati) [inline tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/Locales.txt:11]: are the IBM ones just rearranged (and the lowercase 'ibm' lines added)?

**Comment by srl295** [comment]
> "Caused by: java.lang.IllegalArgumentException: Cannot have default content locale in Locales.txt: google ; nb ; moderate ;"

thanks… will fix and rerun tests.

**Comment by srl295** [comment]
well… turns out orgs "have" to have something in Locales.txt, according to the tests

```
      Error: (StandardCodesTest.java:47) Organization missing Locales.txt information afghan_csa
      Error: (StandardCodesTest.java:47) Organization missing Locales.txt information afghan_mcit
      Error: (StandardCodesTest.java:47) Organization missing Locales.txt information afrigen
      Error: (StandardCodesTest.java:47) Organization missing Locales.txt information bangor_univ
      Error: (StandardCodesTest.java:47) Organization missing Locales.txt information bhutan
      Error: (StandardCodesTest.java:47) Organization missing Locales.txt information india
      Error: (StandardCodesTest.java:47) Organization missing Locales.txt information iran_hci
      Error: (StandardCodesTest.java:47) Organization missing Locales.txt information kendra
      Error: (StandardCodesTest.java:47) Organization missing Locales.txt information kunsill_malti
      Error: (StandardCodesTest.java:47) Organization missing Locales.txt information lao_dpt
      Error: (StandardCodesTest.java:47) Organization missing Locales.txt information mozilla
      Error: (StandardCodesTest.java:47) Organization missing Locales.txt information openinstitute
      Error: (StandardCodesTest.java:47) Organization missing Locales.txt information pakistan
      Error: (StandardCodesTest.java:47) Organization missing Locales.txt information rodakych
      Error: (StandardCodesTest.java:47) Organization missing Locales.txt information welsh_lc
      Error: (StandardCodesTest.java:47) Organization missing Locales.txt information yahoo
```

**Comment by srl295** [comment]
huh, not sure how to fix the following…

```
      Error: (TestCLDRLocaleCoverage.java:204) The following Apple moderate+modern locales were absent from the Cldr moderate+modern locales:[aa, bo, cad, kl, kok, lb, pcm, tg, tt]
        aa      Afar
        bo      Tibetan
        cad     Caddo
        kl      Kalaallisut
        kok     Konkani
        lb      Luxembourgish
        pcm     Nigerian Pidgin
        tg      Tajik
        tt      Tatar
      Error: (TestCLDRLocaleCoverage.java:205) The following Google moderate+modern locales were absent from the Cldr moderate+modern locales:[aa, br, gd, ia, kea, nqo, oc, sc]
        aa      Afar
        br      Breton
        gd      Scottish Gaelic
        ia      Interlingua
        kea     Kabuverdianu
        nqo     N?Ko
        oc      Occitan
        sc      Sardinian
```

**Review by macchiati — COMMENTED** [review]
Looks good to me. @pedberg-icu should also review.

File: tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/Locales.txt (hunk 1 above)
Comment (srl295) [reply tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/Locales.txt:11]: yes

**Comment by srl295** [comment]
> huh, not sure how to fix the following…
> 
> ```
>       Error: (TestCLDRLocaleCoverage.java:204) The following Apple moderate+modern locales were absent from the Cldr moderate+modern locales:[aa, bo, cad, kl, kok, lb, pcm, tg, tt]
>       Error: (TestCLDRLocaleCoverage.java:205) The following Google moderate+modern locales were absent from the Cldr moderate+modern locales:[aa, br, gd, ia, kea, nqo, oc, sc]
> ```

@macchiati do these need to be added to org 'Cldr' and at what level?

**Comment by srl295** [comment]
> well… turns out orgs "have" to have something in Locales.txt, according to the tests
> 
> ```
>       Error: (StandardCodesTest.java:47) Organization missing Locales.txt information afghan_csa
>       Error: (StandardCodesTest.java:47) Organization missing Locales.txt information afghan_mcit
>       Error: (StandardCodesTest.java:47) Organization missing Locales.txt information afrigen
>       Error: (StandardCodesTest.java:47) Organization missing Locales.txt information bangor_univ
>       Error: (StandardCodesTest.java:47) Organization missing Locales.txt information bhutan
>       Error: (StandardCodesTest.java:47) Organization missing Locales.txt information india
>       Error: (StandardCodesTest.java:47) Organization missing Locales.txt information iran_hci
>       Error: (StandardCodesTest.java:47) Organization missing Locales.txt information kendra
>       Error: (StandardCodesTest.java:47) Organization missing Locales.txt information kunsill_malti
>       Error: (StandardCodesTest.java:47) Organization missing Locales.txt information lao_dpt
>       Error: (StandardCodesTest.java:47) Organization missing Locales.txt information mozilla
>       Error: (StandardCodesTest.java:47) Organization missing Locales.txt information openinstitute
>       Error: (StandardCodesTest.java:47) Organization missing Locales.txt information pakistan
>       Error: (StandardCodesTest.java:47) Organization missing Locales.txt information rodakych
>       Error: (StandardCodesTest.java:47) Organization missing Locales.txt information welsh_lc
>       Error: (StandardCodesTest.java:47) Organization missing Locales.txt information yahoo
> ```

For now, I will _leave_ these organizations in (but without the asterisk) but with a warning comment. 

Otherwise i could add them with locale `mul` or somethign else … or change the test to a warning?

**Comment by macchiati** [comment]
We normally have the CLDR locales be a union of the tc members' locales. So
the test is to raise a red flag when that happens.

I'd asked for the tc member locales + CLDR to stay the same as they were,
So something went wrong there.

**Comment by macchiati** [comment]
Hmm. We could include und as a placeholder if the locale is otherwise empty.

**Comment by srl295** [comment]
> We normally have the CLDR locales be a union of the tc members' locales. So the test is to raise a red flag when that happens. 

OK, that's what i thought

> I'd asked for the tc member locales + CLDR to stay the same as they were, So something went wrong there.

Apple voted in Afar (`aa`) in v38 for example. So that's why I added `aa`  to prevent old votes from being deleting.

**Comment by srl295** [comment]
OK, all OK except for the Cldr superset.

**Comment by macchiati** [comment]
One note: when I said CLDR is a union, I meant of the locales being
actively worked on. Certain of the Microsoft locales don't have active
vetters, so we have a special set of exceptions in the tests for that. We
don't want to include those exceptions in the CLDR locale, since it will
muck up other things.


Mark

**Comment by srl295** [comment]
> One note: when I said CLDR is a union, I meant of the locales being actively worked on. Certain of the Microsoft locales don't have active vetters, so we have a special set of exceptions in the tests for that. We don't want to include those exceptions in the CLDR locale, since it will muck up other things. Mark
> […](#)

 I assumed that the Cldr org  _is_ the definition of what's actively worked  on.  Anyway. I think i found the exception list, and will update. I kind of think the upshot is that more review is needed.

**Comment by macchiati** [comment]
Yes. It was very hard to review because everything was reordered; makes it
easy to miss when something is added or removed.

Mark

**Comment by srl295** [comment]
> Yes. It was very hard to review because everything was reordered; makes it easy to miss when something is added or removed. Mark

Agreed. It was hard to write for the same reason, because the original file is not ordered.  Perhaps this file should be kept in sorted order in the future?

Do you need some further analysis, such as an intermediate commit that only reorders the lines?

**Comment by srl295** [comment]
tests pass locally

**Comment by macchiati** [comment]
I think we can keep them in order in the future; we could add a field if an
Org wants a particular order (we like to keep them in particular levels).

Mark

**Comment by srl295** [comment]
It's the SurveyTool tests that are failing, probably due to changed voting restrictions. Will fix

**Comment by macchiati** [comment]
Thanks

**Comment by srl295** [comment]
OK, turns out the ST tests had to be updated to match changes to Locales.txt

**Review by btangmu — APPROVED** [review]

**Comment by srl295** [comment]
@macchiati trying to move this forward. Can still revisit before push to production

**Comment by macchiati** [comment]
What's the ask? Review?

**Comment by macchiati** [comment]
Ah I understand. Yes let's move this forward and we can tweak afterwards.

**Comment by srl295** [comment]
> What's the ask? Review?

I should have said, “I have merged this so that we move forward. It can still be reviewed before the final push to production, if there are any remaining concerns."

**Pull request merged**